src/utils/pivotExceptions.ts

```diff
@@ -0,0 +1,35 @@
+import { extractPointerFromRef, extractSourceFromRef, pointerToPath } from '@stoplight/json';
+import { Dictionary, JsonPath } from '@stoplight/types';
+import { IRunRule } from '../types';
+import { RulesetExceptionCollection } from '../types/ruleset';
+
+export interface IExceptionLocation {
+  source: string;
+  path: JsonPath;
+}
+
+export const pivotExceptions = (
+  exceptions: RulesetExceptionCollection,
+  runRules: Dictionary<IRunRule, string>,
+): Dictionary<IExceptionLocation[], string> => {
+  const dic: Dictionary<IExceptionLocation[], string> = {};
+
+  Object.entries(exceptions).forEach(([location, rules]) => {
+    const pointer = extractPointerFromRef(location)!;
```

Comment:
Should we account for lacking pointers? I'm pretty sure we validate them for the most of the time at the moment, so there won't be a case when there is a local ref provided, but what if that function is used in a different context?

Reply:
@P0lip Fixed